src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisCodeEditorSessionService.java

```diff
@@ -0,0 +1,161 @@
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, session.getExercise(), user);
+        if (!Objects.equals(session.getUser(), user)) {
+            throw new AccessForbiddenException("Iris Code Editor Session", session.getId());
+        }
+    }
+
+    @Override
+    public void checkIsIrisActivated(IrisSession session) {
+        checkIsIrisActivated(castToSessionType(session, IrisCodeEditorSession.class));
+    }
+    
+    private void checkIsIrisActivated(IrisCodeEditorSession ignored) {
+        // Code editor sessions should probably be available for every programming exercise, especially just-created ones
+        // However, we still may want to check something here
+        // FIXME: Await update to Iris settings system
+    }
+    
+    /**
+     * Sends a request containing the current state of the exercise repositories in the code editor
+     * and the entire conversation history to the LLM, and handles the response.
+     *
+     * @param irisSession The session to send the request for
+     */
+    @Override
+    public void requestAndHandleResponse(IrisSession irisSession) {
+        var fromDB = irisSessionRepository.findByIdWithMessagesAndContents(irisSession.getId());
+        if (!(fromDB instanceof IrisCodeEditorSession session)) {
+            throw new BadRequestException("Iris session is not a code editor session");
+        }
+        var exercise = session.getExercise();
+        var params = new HashMap<String, Object>();
+        params.put("chat_history", session.getMessages());
+        params.put("ps", exercise.getProblemStatement());
+        params.put("solution_repo", getRepositoryContents(exercise.getVcsSolutionRepositoryUrl()));
+        params.put("template_repo", getRepositoryContents(exercise.getVcsTemplateRepositoryUrl()));
+        params.put("test_repo", getRepositoryContents(exercise.getVcsTestRepositoryUrl()));
+        
+        // FIXME: Template and model should be be configurable; await settings update
+        irisConnectorService.sendRequest(new IrisTemplate(IrisConstants.CODE_EDITOR_INITIAL_REQUEST), "gpt-4-32k", params)
+                .handleAsync((responseMessage, err) -> {
+                    if (err != null) {
+                        log.error("Error while getting response from Iris model", err);
+                        irisWebsocketService.sendException(session, err.getCause());
+                    }
+                    else if (responseMessage == null) {
+                        log.error("No response from Iris model");
+                        irisWebsocketService.sendException(session, new IrisNoResponseException());
+                    }
+                    else {
+                        var irisMessageSaved = irisMessageService.saveMessage(responseMessage.message(), session, IrisMessageSender.LLM);
+                        irisWebsocketService.sendMessage(irisMessageSaved);
+                    }
+                    return null;
```

Comment:
Optional: 
We could probably extract this into a method that is used by the chat and code editor services.

Reply:
Leaving this out for now. Would require a fair bit of refactoring and we're not sure yet if there is a good reason why all session types should need to handle responses the same way